src/test/test_switches.cpp

```diff
@@ -0,0 +1,315 @@
+/**
+ * This file is part of ExpressLRS
+ * See https://github.com/AlessandroAU/ExpressLRS
+ *
+ * Unit tests for over the air packet encoding, decoding and associated utilities
+ *
+ * Entry point is setup_switches()
+ */
+
+#include <Arduino.h>
+#include <unity.h>
+
+#include "LoRaRadioLib.h"
+#include "CRSF.h"       // has to come after LoraRadioLib.h for R9
+#include "targets.h"
+#include "common.h"
+#include <OTA.h>
+
+// Tests have only been run on TTGO V1, so we have to guard
+// all the testcode to prevent breakages until it's know to do
+// something sensible.
+
+#ifdef TARGET_TTGO_LORA_V1_AS_TX
+
+CRSF crsf;  // need an instance to provide the fields used by the code under test
+
+SX127xDriver Radio; // needed for Radio.TXdataBuffer
+
+/* Check that the round robin works
+ * First call should return 0 for seq switches or 1 for hybrid
+ * Successive calls should increment the next index until wrap
+ * around from 7 to either 0 or 1 depending on mode.
+ */
+void test_round_robin(void) {
+
+    uint8_t expectedIndex=crsf.nextSwitchIndex;
```

Comment:
[Pedantic] Whitespace between operators on multiple line in here

Reply:
yeah, we're going to have to talk about this whitespace thing :)